acceptance/builder_test.go

```diff
@@ -0,0 +1,248 @@
+					"docker",
+					"run",
+					"--rm",
+					"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+					"--env", "CNB_GROUP_PATH=/cnb/group_tomls/always_detect_group.toml",
+					builderImage,
+				)
+				output, err := command.CombinedOutput()
+				h.AssertNotNil(t, err)
+				expected := "failed to parse detect plan"
+				h.AssertStringContains(t, string(output), expected)
+			})
+
+			//TODO: check some output file for this case not except any error message
+			it("empty parse plan.toml file", func() {
+				command := exec.Command(
+					"docker",
+					"run",
+					"--rm",
+					"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+					"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/empty_plan.toml",
+					"--env", "CNB_GROUP_PATH=/cnb/group_tomls/always_detect_group.toml",
+					builderImage,
+				)
+				_, err := command.CombinedOutput()
+				//print(string(output), err)
+				h.AssertNil(t, err)
+				//expected := "failed to read buildpack order file"
+				//h.AssertStringContains(t, string(output), expected)
+			})
+
+			it("invalid parse plan.toml file", func() {
+				command := exec.Command(
+					"docker",
+					"run",
+					"--rm",
+					"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+					"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/wrong_plan.toml",
+					"--env", "CNB_GROUP_PATH=/cnb/group_tomls/always_detect_group.toml",
+					builderImage,
+				)
+				output, err := command.CombinedOutput()
+				h.AssertNotNil(t, err)
+				expected := "failed to parse detect plan: Near line"
+				h.AssertStringContains(t, string(output), expected)
+			})
+
+		})
+	})
+
+	when("PlaceHolders need to replace with defaults", func() {
+
+		// .../cmd/lifecycle/builder.go#48-50
+		when("groupPath is equals to PlaceHolder groupPath", func() {
+			it("will replace placeholder with default groupPath", func() {
+				command := exec.Command(
+					"docker",
+					"run",
+					"--rm",
+					"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+					"--env", "CNB_GROUP_PATH=<layers>/group.toml",
+					"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/always_detect_plan.toml",
+					builderImage,
+				)
+				output, err := command.CombinedOutput()
+				h.AssertNotNil(t, err)
+				expected := "failed to read buildpack group: open /layers/group.toml"
+				h.AssertStringContains(t, string(output), expected)
+			})
+		})
+
+		// .../cmd/lifecycle/builder.go#52-54
+		when("planPath is equals to PlaceHolder planPath", func() {
+			it("will replace placeholder with default planPath", func() {
+				command := exec.Command(
+					"docker",
+					"run",
+					"--rm",
+					"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+					"--env", "CNB_PLAN_PATH=<layers>/plan.toml",
+					"--env", "CNB_GROUP_PATH=/cnb/group_tomls/always_detect_group.toml",
+					builderImage,
+				)
+				output, err := command.CombinedOutput()
+				h.AssertNotNil(t, err)
+				expected := "failed to parse detect plan: open /layers/plan.toml"
+				h.AssertStringContains(t, string(output), expected)
+			})
+		})
+	})
+
```

Comment:
I think it's important to add some tests for good path that will test https://github.com/buildpacks/lifecycle/blob/main/cmd/lifecycle/builder.go#L78-L112 and more specifically will test https://github.com/buildpacks/lifecycle/blob/main/cmd/lifecycle/builder.go#L97 and its output https://github.com/buildpacks/lifecycle/blob/main/cmd/lifecycle/builder.go#L108-L110

Reply:
Creating tests for these 3 scenarios sounds great. I will create these tests for the next step.

Reply:
i think i will parse `/layers/config/metadata.toml` then assert content according to my buildpack and other inputs. Please add any other suggestions you can think of.

Reply:
please check this commit, https://github.com/buildpacks/lifecycle/pull/648/commits/5840857629d7f90d3ffe40bbdeab38fa26dd0d5d